shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLUTTERPLUGIN_H_
+#define FLUTTER_FLUTTERPLUGIN_H_
+
+#include "FlutterBinaryMessenger.h"
+#include "FlutterChannels.h"
+#include "FlutterCodecs.h"
+
+NS_ASSUME_NONNULL_BEGIN
+@protocol FlutterPluginRegistrar;
+
+@protocol FlutterPlugin <NSObject>
+@required
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar;
+@optional
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result;
+- (BOOL)application:(UIApplication*)application didFinishLaunchingWithOptions:(NSDictionary*)launchOptions;
```

Comment:
Aren't these just `UIApplicationDelegate` protocol methods? Instead of listing them all here again, you can make the `FlutterPlugin` just inherit the `UIApplicationDelegate`.